File: crates/opengrok-server/src/gateway/fixtures/mock-sample.js
// Mock fixture: enough JavaScript to judge the highlighter.
import { readFile } from "node:fs/promises";

const FIXTURE_DIR = "/tmp/opengrok-mock-fixtures";
const retries = 3;

/**
 * Read a fixture and describe it. Template literal, async/await,
 * optional chaining and a regex all appear on purpose.
 */
export async function describe(name) {
  const path = `${FIXTURE_DIR}/${name}`;
  const bytes = await readFile(path);
  const kind = /\.(png|mp4|pdf)$/.exec(name)?.[1] ?? "text";
  return { name, size: bytes.length, kind, ok: bytes.length > 0 };
}

for (let i = 0; i < retries; i++) {
  if (i === retries - 1) console.log("done", i);
}

File: crates/opengrok-server/src/gateway/fixtures/mock_sample.rs
//! Mock fixture: enough Rust to judge the highlighter.
use std::path::PathBuf;

const FIXTURE_DIR: &str = "/tmp/opengrok-mock-fixtures";

/// A fixture on disk, described. Lifetimes, generics, a macro and a
/// match all appear on purpose.
#[derive(Debug, Clone)]
pub struct Fixture<'a> {
    pub name: &'a str,
    pub size: usize,
}

pub fn describe<'a>(name: &'a str, bytes: &[u8]) -> Fixture<'a> {
    let kind = match name.rsplit_once('.') {
        Some((_, "png" | "mp4" | "pdf")) => "binary",
        Some(_) | None => "text",
    };
    println!("{name}: {} bytes ({kind})", bytes.len());
    Fixture { name, size: bytes.len() }
}

fn path_of(name: &str) -> PathBuf {
    PathBuf::from(FIXTURE_DIR).join(name)
}
